### PR TITLE
[RELAY] Fix type info after mutation in simplify inference

### PR DESCRIPTION
**Pull request by vinx13**
`SimplifyInference` throws an assert error if the function contains multiple batch norm. (See the updated test case, previously it calls `simplify_inference` each time a new batch norm is added) The new expression after mutation doesn't have checked type, so when you unpack the next batch norm, you cannot get the type of the input data.

This patch fix this issue by mapping data of batch norm to its type before mutation when visiting a `call(batch_norm)`. This works since we don't change the shape of data.

See also https://discuss.tvm.ai/t/relay-type-inference-for-this-expr-has-not-completed-during-simplify-inference/1095

Please review @ZihengJiang @tqchen @masahi @yzhliu

### CODE REVIEW

File: src/relay/pass/simplify_inference.cc [hunk 1]
```diff
@@ -32,9 +33,9 @@ Expr BatchNormToInferUnpack(const Attrs attrs,
   }
 
   int axis = param->axis;
-  const auto* tdata = data->type_as<TensorTypeNode>();
-  scale = ExpandBiasToMatchAxis(scale, tdata->shape.size(), {axis});
-  shift = ExpandBiasToMatchAxis(shift, tdata->shape.size(), {axis});
+  auto ndim = tdata.as<TensorTypeNode>()->shape.size();
```
Comment (tqchen) [inline src/relay/pass/simplify_inference.cc]: be safe, CHECK data.as<TensorTypeNode>() is not null

**Comment by tqchen** [comment]
cc @jroesch

**Comment by tqchen** [comment]
Thanks, @vinx13 ! this is now merged.
